salt/etc-hosts/hosts.jinja

```diff
@@ -9,7 +11,7 @@
 	{%- else %}
 		{%- set ip = addrlist|first %}
 	{%- endif %}
-{{ ip }} {{ admin_id }} {{ admin_id }}.{{ pillar['internal_infra_domain'] }}
+{{ ip }} {{ admin_id }} {{ admin_id }}.{{ pillar['internal_infra_domain'] }} {{ hostnames[admin_id] }}
```

Comment:
As we have no requirement that hostnames are unique, this and the lines below will fail on currently supported environments due to placing duplicated entries for the duplicated names into /etc/hosts.

Reply:
This is good point, I will think how to check if all names are unique.